util/dvsim/NcLauncher.py

```diff
@@ -86,49 +85,49 @@ def _do_launch(self):
-            log.log(VERBOSE, '[Executing]:\n{}\n\n'.format(self.deploy.cmd))
+            cmd_arr = self.get_submit_cmd(['-I'])
+            buf_size = 1
+            std_out = subprocess.PIPE
+            std_err = subprocess.STDOUT
+        else:
+            cmd_arr = self.get_submit_cmd(['-wl'])
+            buf_size = 4096
+            std_out = fd
+            std_err = fd
+
+        log.log(VERBOSE, '[Submitting]:\n{}\n\n'.format(' '.join(cmd_arr)))
+
+        try:
             self.process = subprocess.Popen(cmd_arr,
-                                            stdin=None,
-                                            stdout=None,
-                                            stderr=subprocess.STDOUT,
-                                            # string mode
+                                            bufsize=buf_size,
                                             universal_newlines=True,
+                                            pass_fds = [fd],
+                                            stdout=std_out,
+                                            stderr=std_err,
                                             env=exports,
                                             cwd=self.deploy.odir)
-
-            # Wait until the process exits
-            self.process.wait()
-        else:
-            try:
-                interactive_flags = ['-nolog', '-wl']
-                cmd_arr = self.get_submit_cmd(interactive_flags)
-                # Using file open instead of with open as
-                # it is being using in the subprocess.Popen call
-                # which returns immediately after launching the cmd
-                # and we want the file to remain open throughout the process
-                f = open(self.deploy.get_log_path(),
-                         'w',
-                         encoding='UTF-8',
-                         errors='surrogateescape')
-                f.write('[Executing]:\n{}\n\n'.format(self.deploy.cmd))
-                f.flush()
-                self.process = subprocess.Popen(cmd_arr,
-                                                bufsize=4096,
-                                                universal_newlines=True,
-                                                stdout=f,
-                                                stderr=f,
-                                                env=exports,
-                                                cwd=self.deploy.odir)
-            except subprocess.SubprocessError as e:
-                raise LauncherError(f'IO Error: {e}\n'
-                                    f'See {self.deploy.get_log_path()}')
-            finally:
-                self._close_process()
+            if self.deploy.sim_cfg.interactive:
+                for line in self.process.stdout:
+                    os.write(fd, line.encode())
+                    sys.stdout.write(line)
+                # Wait until the process exits
+                self.process.wait()
```

Comment:
Does this `wait` call ever do anything? I guess it only matters if the subprocess happens to close its `stdout` (ending the loop above)? If I've understood properly, maybe it makes sense to leave a note in the comment to stop the next person wondering :-)

Reply:
done. added comment